sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveWithCTE.scala

```diff
@@ -79,6 +79,7 @@ object ResolveWithCTE extends Rule[LogicalPlan] {
                     cteDef.id,
                     anchor,
                     rewriteRecursiveCTERefs(recursion, anchor, cteDef.id, None),
+                    anchor.output.map(_.newInstance().exprId),
```

Comment:
nit: we can add a new `apply` method in `object UnionLoop`, which takes the same parameters as before, and pass `anchor.output.map(_.newInstance().exprId)` additionally to construct `UnionLoop`